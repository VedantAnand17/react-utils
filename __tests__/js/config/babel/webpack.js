import getConfig from '../../../../config/babel/webpack';

test('Module exports', () => {
  expect(getConfig).toMatchSnapshot();
});

describe('"development" environment', () => {
  const babel = { env: () => 'development' };

  test('Default config', () => {
    expect(getConfig(babel)).toMatchSnapshot();
  });

  test('"noStyling" variation', () => {
    expect(getConfig(babel, { noStyling: true })).toMatchSnapshot();
  });
});

describe('"production" environment', () => {
  const babel = { env: () => 'production' };

  test('Default config', () => {
    expect(getConfig(babel)).toMatchSnapshot();
  });

  test('"noStyling" variation', () => {
    expect(getConfig(babel, { noStyling: true })).toMatchSnapshot();
  });
});
