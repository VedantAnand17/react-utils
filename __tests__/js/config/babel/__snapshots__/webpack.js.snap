// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"development" environment "noStyling" variation 1`] = `
{
  "plugins": [
    [
      "module-resolver",
      {
        "extensions": [
          ".js",
          ".jsx",
        ],
        "root": [
          "./src/shared",
          "./src",
        ],
      },
    ],
    "@babel/transform-runtime",
    "react-refresh/babel",
  ],
  "presets": [
    [
      "@babel/env",
      {
        "targets": "defaults or chrome >= 69",
      },
    ],
    [
      "@babel/react",
      {
        "runtime": "automatic",
      },
    ],
    "@dr.pogodin/babel-preset-svgr",
  ],
}
`;

exports[`"development" environment Default config 1`] = `
{
  "plugins": [
    [
      "module-resolver",
      {
        "extensions": [
          ".js",
          ".jsx",
        ],
        "root": [
          "./src/shared",
          "./src",
        ],
      },
    ],
    "@babel/transform-runtime",
    [
      "@dr.pogodin/react-css-modules",
      {
        "autoResolveMultipleImports": true,
        "filetypes": {
          ".scss": {
            "syntax": "postcss-scss",
          },
        },
        "generateScopedName": [Function],
      },
    ],
    "react-refresh/babel",
  ],
  "presets": [
    [
      "@babel/env",
      {
        "targets": "defaults or chrome >= 69",
      },
    ],
    [
      "@babel/react",
      {
        "runtime": "automatic",
      },
    ],
    "@dr.pogodin/babel-preset-svgr",
  ],
}
`;

exports[`"production" environment "noStyling" variation 1`] = `
{
  "plugins": [
    [
      "module-resolver",
      {
        "extensions": [
          ".js",
          ".jsx",
        ],
        "root": [
          "./src/shared",
          "./src",
        ],
      },
    ],
    "@babel/transform-runtime",
  ],
  "presets": [
    [
      "@babel/env",
      {
        "targets": "defaults or chrome >= 69",
      },
    ],
    [
      "@babel/react",
      {
        "runtime": "automatic",
      },
    ],
    "@dr.pogodin/babel-preset-svgr",
  ],
}
`;

exports[`"production" environment Default config 1`] = `
{
  "plugins": [
    [
      "module-resolver",
      {
        "extensions": [
          ".js",
          ".jsx",
        ],
        "root": [
          "./src/shared",
          "./src",
        ],
      },
    ],
    "@babel/transform-runtime",
    [
      "@dr.pogodin/react-css-modules",
      {
        "autoResolveMultipleImports": true,
        "filetypes": {
          ".scss": {
            "syntax": "postcss-scss",
          },
        },
        "generateScopedName": [Function],
      },
    ],
  ],
  "presets": [
    [
      "@babel/env",
      {
        "targets": "defaults or chrome >= 69",
      },
    ],
    [
      "@babel/react",
      {
        "runtime": "automatic",
      },
    ],
    "@dr.pogodin/babel-preset-svgr",
  ],
}
`;

exports[`Module exports 1`] = `[Function]`;
